request_logging/middleware.py

```diff
@@ -141,9 +191,19 @@ def _should_log_route(self, request):
         elif hasattr(view, 'view_class'):
             # This is for django class-based views
             func = getattr(view.view_class, method, None)
-        no_logging = getattr(func, NO_LOGGING_ATTR, False)
-        no_logging_msg = getattr(func, NO_LOGGING_MSG_ATTR, None)
-        return no_logging, no_logging_msg
+
+        return func
+
+    def _should_log_route(self, request):
+        func = self._get_api_func(request)
+        if func in OPT_INTO_LOGGING_FUNCS:
+            return ShouldLogRoute(log_route=True, skip_reason=None, silent=False)
+
+        if func in NO_LOGGING_FUNCS:
+            no_logging = NO_LOGGING_FUNCS[func]
+            return ShouldLogRoute(log_route=False, skip_reason=no_logging.message, silent=no_logging.silent)
+
+        return ShouldLogRoute(log_route=self.logging_opt_in_defaults[REQUEST](request), skip_reason=None, silent=False)
```

Comment:
I couldn't really come up with a better way to handle the silent case than propagating a flag forward. It definitely feels a little leaky but it works with current behavior